generate_text_instructions.py

```diff
@@ -0,0 +1,149 @@
+import csv
+import random
+import sys
+
+import template_file_reader as reader
+
+FLIP_PERCENTAGE = 0.3
+
+def main():
+    args = get_args()
+
+    if not args:
+        return
+
+    colors = load_colors()
+    letters = load_letters()
+
+    data = build_data(args['num-to-generate'], colors, letters)
+
+    data = expand_data(data, len(colors), len(letters))
+
+    write_csv(args['neural-infile'], data['text'])
+    write_csv(args['neural-flips-outfile'], data['flipped'])
+    write_csv(args['neural-color-outfile'], data['colors'])
+    write_csv(args['neural-letter-outfile'], data['letters'])
+
+def get_args():
+    try:
+        assert len(sys.argv) == 6
+
+        args = {}
+
+        args['num-to-generate'] = int(sys.argv[1])
+        args['neural-infile'] = sys.argv[2]
+        args['neural-flips-outfile'] = sys.argv[3]
+        args['neural-color-outfile'] = sys.argv[4]
+        args['neural-letter-outfile'] = sys.argv[5]
+
+        return args
+    except:
+        print('Call as "python generate_text_instructions.py <num-to-generate> <neural-infile> <neural-flips-outfile> <neural-color-outfile> <neural-letter-outfile>"')
+
+def load_colors():
+    with open('colors.txt') as colors_file:
+        return list(map(lambda color: color.replace('\n', ''), colors_file.readlines()))
+
+def load_letters():
+    letter_nums = list(range(ord('A'), ord('H') + 1))
+    return list(map(lambda i: chr(i), letter_nums))
+
+def build_data(num_to_generate, colors, letters):
+    data = {
+        'text': [],
+        'flipped': [],
+        'colors': [],
+        'letters': []
+    }
+
+    for _ in range(num_to_generate):
+        if should_flip():
+            template_func = reader.read_flip_template
+            data['flipped'] += [True]
+        else:
+            template_func = reader.read_move_template
+            data['flipped'] += [False]
```

Comment:
I think it would technically be faster to use `append` here, but this makes up a pretty small amount of the computation time running the neural network pipeline